.github/workflows/verify_pr.yml

```diff
@@ -0,0 +1,26 @@
+name: Verify Pull Request
```

Comment:
Here, I've merged the test pipeline with linter pipeline, which was introduced lately by @renanvalentin, as well as added a few more checks. I think this is as per our discussion to reduce the amount of work done by CI - when those two were running separately, we'd spend 75% of the time on installing `node_modules`. I'd like to start a discussion here and see which approach works better for us, separate, more granular pipelines, or one which checks everything?

Reply:
Guys, at the moment unit tests take less than 30 seconds to execute. Any gain from parallelization will be negligible, especially that we're meant to run this locally before pushing. The only potential win would be by splitting `yarn check:all` but I feel this is too much outside of scope of this PR, and adds another layer of complexity as we would need to share files across the jobs (either by mounting a shared volume or by uploading and then downloading artifacts). I think such improvements definitely make sense but when the pipeline takes longer than 5 minutes to execute.

Anyway I feel this PR introduces a significant improvement in terms of reducing the build time by introducing cache for node_modules - installing packages now takes 15 seconds rather than 1.5 minutes:

![image](https://user-images.githubusercontent.com/6684554/156023663-827639b0-0169-4127-be5c-d0ac65786411.png)